src/editor/controller/editor_controller.cpp

```diff
@@ -877,6 +877,10 @@ bool editor_controller::do_execute_command(const hotkey::ui_command& cmd, bool p
 		{
 			initialize_addon_if_empty();
 
+			if (current_addon_id_.empty()) {
```

Comment:
Probably rather for a follow-up refactoring but it seems like the logic for "init was done or not" better belongs inside `initialize_addon_if_empty`. It should return bool IMO.

Reply:
@soliton- Implemented, please see if https://github.com/wesnoth/wesnoth/pull/9273/commits/c858bd1c7fbe8546c8a6809d528f39488c6c9cd2 is fine?